### ISSUE TITLE
Finetuning YOLOWORLD on custom datasets

### ISSUE THREAD

**Issue by toobatehreem** [ultralytics/ultralytics#17371]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Is it possible to fine-tune YOLOWORLD on a training dataset with ten classes and validate the fine-tuned model on a separate validation dataset with twenty different classes?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @toobatehreem, thank you for reaching out to Ultralytics 🚀! We're here to assist, and an Ultralytics engineer will also be with you soon.

For new users, we suggest checking out our [Docs](https://docs.ultralytics.com) where you can find extensive usage examples for both [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/). Many common queries may already be addressed there.

If this is a custom training ❓ Question, please provide more information to assist us in helping you better. This could include details about your dataset, training logs, and the specific steps you've already taken. Additionally, ensure you are following the guidelines in our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, hop into our [Discord](https://ultralytics.com/discord) 🎧. Prefer in-depth discussions? Visit [Discourse](https://community.ultralytics.com). Or engage with fellow enthusiasts on our [Subreddit](https://reddit.com/r/ultralytics) to share insights.

## Upgrade

Ensure you're using the latest version of the `ultralytics` package with all the necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) installed in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be executed in several up-to-date verified environments, complete with all dependencies such as [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) already set up:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, it indicates that all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
@toobatehreem yes, you can fine-tune YOLO-World on a training dataset with ten classes and validate it on a separate validation dataset with twenty different classes by setting custom prompts for each dataset.

**Comment by toobatehreem** [comment]
> @toobatehreem yes, you can fine-tune YOLO-World on a training dataset with ten classes and validate it on a separate validation dataset with twenty different classes by setting custom prompts for each dataset.

Can you please give an example of this.

**Comment by glenn-jocher** [comment]
@toobatehreem certainly! You can use the `set_classes` method to specify different classes for training and validation datasets in YOLO-World.
